cmd/topicctl/subcmd/rebalance.go

```diff
@@ -151,55 +159,97 @@ func rebalanceRun(cmd *cobra.Command, args []string) error {
 		return err
 	}
 
-	// iterate through each topic config and initiate rebalance
-	topicConfigs := []config.TopicConfig{}
-	topicErrorDict := make(map[string]error)
-	for _, topicFile := range topicFiles {
-		// do not consider invalid topic yaml files for rebalance
-		topicConfigs, err = config.LoadTopicsFile(topicFile)
+	tmpDir := ""
+	existingConfigFiles := []string{}
+	if rebalanceConfig.generateConfig {
+		// make set of existing files
+		err := processTopicFiles(topicFiles, func(topicConfig config.TopicConfig, topicFile string) error {
+			existingConfigFiles = append(existingConfigFiles, topicConfig.Meta.Name)
+			return nil
+		})
 		if err != nil {
-			log.Errorf("Invalid topic yaml file: %s", topicFile)
-			continue
+			return err
 		}
 
-		for _, topicConfig := range topicConfigs {
-			// topic config should be consistent with the cluster config
-			if err := config.CheckConsistency(topicConfig.Meta, clusterConfig); err != nil {
-				log.Errorf("topic file: %s inconsistent with cluster: %s", topicFile, clusterConfigPath)
-				continue
-			}
+		// create temp dir
+		tmpDir = rebalanceConfig.pathPrefix
+		if rebalanceConfig.pathPrefix[len(rebalanceConfig.pathPrefix) - 1] != '/' {
+			tmpDir += "/"
+		}
+		tmpDir += "tmp/"
+		err = os.Mkdir(tmpDir, 0755)
+		if err != nil {
+			return err
+		}
+		log.Infof("tmp output path: %v", tmpDir)
 
-			log.Infof(
-				"Rebalancing topic %s from config file %s with cluster config %s",
-				topicConfig.Meta.Name,
-				topicFile,
-				clusterConfigPath,
-			)
-
-			topicErrorDict[topicConfig.Meta.Name] = nil
-			rebalanceTopicProgressConfig := util.RebalanceTopicProgressConfig{
-				TopicName:          topicConfig.Meta.Name,
-				ClusterName:        clusterConfig.Meta.Name,
-				ClusterEnvironment: clusterConfig.Meta.Environment,
-				ToRemove:           rebalanceConfig.brokersToRemove,
-				RebalanceError:     false,
-			}
-			if err := rebalanceApplyTopic(ctx, topicConfig, clusterConfig, adminClient); err != nil {
-				topicErrorDict[topicConfig.Meta.Name] = err
-				rebalanceTopicProgressConfig.RebalanceError = true
-				log.Errorf("topic: %s rebalance failed with error: %v", topicConfig.Meta.Name, err)
-			}
+		topicsToExclude := ".^"
+		if len(existingConfigFiles) > 0 {
+			topicsToExclude = strings.Join(existingConfigFiles, "|")
```

Comment:
Looks like this is expected to be a regexp which makes this tricky. For example, if a topic named `foo` had an existing config file, but another topic `foo-dlq` did not, it looks like this would match the latter for exclusion when it shouldn't be?

Example of topics:
```
foo (has config)
foo-dlq (does not)
bar (does not)
```
Then computed exclude regexp would be the string `foo` which would match both foo and foo-dlq and skip bootstrapping of foo-dlq and only bootstraps `bar`